libsolidity/ast/AST.cpp

```diff
@@ -432,8 +432,13 @@ string Scopable::sourceUnitName() const
 
 bool VariableDeclaration::isLValue() const
 {
-	// External function parameters and constant declared variables are Read-Only
-	return !isExternalCallableParameter() && !m_isConstant;
+	// Constant declared variables are Read-Only
+	if (m_isConstant)
+		return false;
+	// External function arguments of reference type are Read-Only
+	if (isExternalCallableParameter() && dynamic_cast<ReferenceType const*>(type()))
```

Comment:
Note that mappings can't be external arguments (and assigning to them generally produces an explicit error elsewhere) - did I miss anything else here?

Reply:
It should not be possible to assign to mappings regardless of their storage type, but can you try this?
```
library L {
function f(mapping(uint=>uint) storage) external {}
}
```

Reply:
I added a test case: https://github.com/ethereum/solidity/blob/e8c08e69b2dcb23376f233971fcfec280dab9704/test/libsolidity/syntaxTests/lvalues/library_mapping.sol

It used to emit two errors and now only emits one, but that's actually nicer.